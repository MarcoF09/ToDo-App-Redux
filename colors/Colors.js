export const Colors = {
  black: "black",
  pink: "#ff197b",
  customBlue: "#1f86ff",
  lightGrey: "#f0f0f0",
  white: "white",
  softGrey: "#959595",
  blue: "blue"
};
