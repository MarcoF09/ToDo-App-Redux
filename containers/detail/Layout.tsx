import React, { FunctionComponent } from 'react';
import { StatusBar, Text, View } from 'react-native';
import { Colors } from '../../colors/Colors';
import { Button } from '../../components/button';
import { Strings } from '../../strings';
import { styles } from './styles';
import { LayoutProps } from './types';

export const Layout: FunctionComponent<LayoutProps> = ({
  markAsDone,
  todo,
  navigation,
  markAsNotDone
}: LayoutProps) => {
  return (
    <View style={styles.container}>
      <StatusBar backgroundColor={Colors.customBlue} />
      <View style={styles.topButtonContainer}>
        <Button
          text={Strings.notDone}
          onPress={() => {
            markAsNotDone(todo, navigation.getParam('index'));
            navigation.pop();
          }}
        />
      </View>
      <Text style={styles.primaryText}>
        {todo[navigation.getParam('index')].title}
      </Text>
      <Text style={styles.secondaryText}>
        {todo[navigation.getParam('index')].description}
      </Text>
      <Button
        text={Strings.markAsDone}
        onPress={() => {
          markAsDone(todo, navigation.getParam('index'));
          navigation.pop();
        }}
      />
    </View>
  );
};
