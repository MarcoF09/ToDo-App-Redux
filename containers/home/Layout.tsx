import React, { FunctionComponent } from 'react'
import { FlatList, StatusBar, View } from 'react-native'
import { Colors } from '../../colors/Colors'
import { renderFooter } from './renderFooter'
import { renderItem } from './renderItem'
import { LayoutProps } from './types'

export const Layout: FunctionComponent<LayoutProps> = ({
  todo,
  navigation,
  clearAllDone,
  moveAnimation,
  deleteItem,
  changeCheckBoxState
}: LayoutProps) => {
  return (
    <View testID="homeLayout">
      <StatusBar backgroundColor={Colors.customBlue} />
      <FlatList
        data={todo}
        keyExtractor={(item, index) => index.toString()}
        renderItem={item =>
          renderItem(
            item,
            navigation,
            todo,
            moveAnimation,
            deleteItem,
            changeCheckBoxState
          )
        }
        ListFooterComponent={() => renderFooter(todo, clearAllDone)}
      />
    </View>
  )
}
