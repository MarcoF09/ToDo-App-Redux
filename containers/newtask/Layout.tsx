import React, { FunctionComponent } from 'react'
import { StatusBar, TextInput, View } from 'react-native'
import { Colors } from '../../colors/Colors'
import { Strings } from '../../strings'
import { styles } from './styles'
import { LayoutProps } from './types'

export const Layout: FunctionComponent<LayoutProps> = ({
  firstInputColor,
  secondInputColor,
  onFocusTitle,
  onBlurTitle,
  onFocusDescription,
  onBlurDescription,
  onChangeTitle,
  onChangeDescription
}: LayoutProps) => {
  return (
    <View style={styles.container} testID="layout">
      <StatusBar backgroundColor={Colors.customBlue} />
      <TextInput
        testID="title"
        onFocus={() => onFocusTitle()}
        onBlur={() => onBlurTitle()}
        style={{
          borderBottomWidth: 1,
          borderBottomColor: firstInputColor,
          fontSize: 36,
          color: Colors.black,
          fontFamily: 'SourceSansPro-Regular'
        }}
        placeholder={Strings.titlePlaceHolder}
        multiline={false}
        onChangeText={text => onChangeTitle(text)}
      />
      <TextInput
        testID="description"
        onFocus={() => onFocusDescription()}
        onBlur={() => onBlurDescription()}
        style={{
          borderBottomWidth: 1,
          borderBottomColor: secondInputColor,
          textAlignVertical: 'top',
          height: 90,
          fontFamily: 'SourceSansPro-Regular',
          fontSize: 14
        }}
        placeholder={Strings.descriptionPlaceHolder}
        multiline={true}
        numberOfLines={5}
        onChangeText={text => onChangeDescription(text)}
      />
    </View>
  )
}
