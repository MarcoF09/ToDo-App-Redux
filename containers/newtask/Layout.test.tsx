import { Colors } from 'colors/Colors'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'
import { Layout } from './Layout'

const createTestProps = (props: object) => ({
  navigation: {
    navigate: jest.fn(),
    getParam: () => 0
  },
  ...props
})

describe('should render new task screen', () => {
  const mockedFunction = jest.fn()
  let props: any
  const title = ''
  const description = ''
  it('renders correctly', () => {
    props = createTestProps({ mockedFunction, title, description })
    const component = shallow(
      <Layout
        firstInputColor={Colors.blue}
        secondInputColor={Colors.blue}
        onFocusTitle={props.mockedFunction}
        onBlurTitle={props.mockedFunction}
        onFocusDescription={props.mockedFunction}
        onBlurDescription={props.mockedFunction}
        onChangeTitle={props.mockedFunction}
        onChangeDescription={props.mockedFunction}
        addToDo={props.mockedFunction}
        navigation={props.navigation}
        title={props.title}
        description={props.description}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
