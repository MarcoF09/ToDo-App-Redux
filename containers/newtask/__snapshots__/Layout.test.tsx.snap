// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render new task screen renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#f0f0f0",
      "borderWidth": 1,
      "padding": 16,
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#1f86ff"
    showHideTransition="fade"
  />
  <TextInput
    allowFontScaling={true}
    multiline={false}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Task title"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "blue",
        "borderBottomWidth": 1,
        "color": "black",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 36,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    multiline={true}
    numberOfLines={5}
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Task description"
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "blue",
        "borderBottomWidth": 1,
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 14,
        "height": 90,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
